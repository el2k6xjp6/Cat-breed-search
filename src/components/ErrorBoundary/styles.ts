import styled from "styled-components";

export const Container = styled.div`
  height: 100vh;
  width: 100vw;
  display: grid;
  place-items: center;
`;

export const ErrorContainer = styled.div`
  width: 400px;
`;

export const ErrorImage = styled.img`
  width: 300px;
  height: 300px;
  margin: 0 50px;
`;

export const ErrorTitle = styled.h1`
  text-align: center;
`;
