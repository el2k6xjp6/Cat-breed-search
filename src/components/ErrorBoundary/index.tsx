import { Component, ErrorInfo, ReactNode } from "react";
import { Container, ErrorContainer, ErrorImage, ErrorTitle } from "./styles";
import ErrorImg from "../../assets/Error.svg";

interface Props {
  children: ReactNode;
}
interface State {
  hasError: boolean;
}

class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error: Error) {
    return { hasError: true, error: error };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.error("Uncaught error:", error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      return (
        <Container>
          <ErrorContainer>
            <ErrorImage src={ErrorImg} alt="Error" />
            <ErrorTitle>Something went wrong.</ErrorTitle>
          </ErrorContainer>
        </Container>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
