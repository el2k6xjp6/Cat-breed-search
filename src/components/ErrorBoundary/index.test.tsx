import ErrorBoundary from ".";
import renderer from "react-test-renderer";

describe("ErrorBoundary component", () => {
  test("snapshot without error", () => {
    const tree = renderer
      .create(
        <ErrorBoundary>
          <div />
        </ErrorBoundary>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with error", () => {
    const ThrowError = () => {
      throw new Error("test");
    };
    const tree = renderer
      .create(
        <ErrorBoundary>
          <ThrowError />
        </ErrorBoundary>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
