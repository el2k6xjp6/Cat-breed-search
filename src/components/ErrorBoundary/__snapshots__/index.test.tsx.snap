// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component snapshot with error 1`] = `
<div
  className="sc-bdvvtL bKoqVk"
>
  <div
    className="sc-gsDKAQ ecILOD"
  >
    <img
      alt="Error"
      className="sc-dkPtRN fNFRLD"
      src="Error.svg"
    />
    <h1
      className="sc-hKwDye iJFtxk"
    >
      Something went wrong.
    </h1>
  </div>
</div>
`;

exports[`ErrorBoundary component snapshot without error 1`] = `<div />`;
